ui/src/main/webapp/app/sidenav/sidenav-component.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017-2018 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+angular.module('app').component('sidenav', sidenav());
+
+function sidenav() {
+    return {
+        templateUrl: 'app/sidenav/sidenav.html',
+        controller: SideNavController,
+        controllerAs: 'ctrl'
+    };
+}
+
+function SideNavController(navigation, config) {
+    var vm = this;
+
+    vm.pages = [
```

Comment:
Would it be possible to use https://docs.angularjs.org/api/ngRoute/service/$route#routes to create this on the fly so if an admin wants to only show a table view, they only have to do it in one place - The route config.